### PR TITLE
cephfs-mirror: disallow adding a active peer back to source

### PR DESCRIPTION
**Pull request by vshankar**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/cephfs/c_cephfs.pxd [hunk 1]
```diff
@@ -46,6 +46,7 @@ cdef extern from "cephfs/libcephfs.h" nogil:
     void ceph_shutdown(ceph_mount_info *cmount)
 
     int ceph_getaddrs(ceph_mount_info* cmount, char** addrs)
+    int64_t ceph_get_fs_cid(ceph_mount_info *cmount)
```
Comment (batrick) [inline src/pybind/cephfs/c_cephfs.pxd:49]:
```suggestion
    int64_t ceph_get_fscid(ceph_mount_info *cmount)
```

fscid == file system cluster id (common acronym in cephfs docs/api)

**Review by batrick — CHANGES_REQUESTED** [review]
otherwise lgtm

File: src/pybind/cephfs/c_cephfs.pxd (hunk 1 above)
Comment (vshankar) [reply src/pybind/cephfs/c_cephfs.pxd:49]: Hmmm . However, libcephfs already has this interface:

```
extern "C" int64_t ceph_get_fs_cid(struct ceph_mount_info *cmount)
```

Comment (batrick) [reply src/pybind/cephfs/c_cephfs.pxd:49]: oh, then crap. Keep it as is then.

**Comment by batrick** [comment]
LGTM after rebase

**Comment by vshankar** [comment]
@batrick rebased and updated

**Review by batrick — APPROVED** [review]

File: src/pybind/cephfs/cephfs.pyx [hunk 2]
```diff
@@ -481,6 +481,17 @@ cdef class LibCephFS(object):
...
 
+    def get_fscid(self):
+        """
+        Return the file system id for this fs client.
+        """
+        self.require_state("mounted")
+        with nogil:
+            ret = ceph_get_fs_cid(self.cluster)
+            if ret < 0:
+                raise make_ex(ret, "error fetching fscid")
```
Comment (batrick) [inline src/pybind/cephfs/cephfs.pyx:492]:
```suggestion
        if ret < 0:
            raise make_ex(ret, "error fetching fscid")
```

to fix

https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/54571//consoleFull

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/pybind/cephfs/cephfs.pyx (hunk 2 above)
Comment (vshankar) [reply src/pybind/cephfs/cephfs.pyx]: thx. done!

**Comment by vshankar** [comment]
jenkins test make check

**Comment by batrick** [comment]
needs rebased

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2021-June-16

**Pull request merged**